typescript/packages/plugins/renzo/src/renzo.service.ts

```diff
@@ -0,0 +1,254 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { RENZO_ABI } from "./abi/renzo";
+import { EZETH_ABI } from "./abi/ezeth";
+import { ERC20_ABI } from "./abi/erc20";
+import { getRenzoAddresses } from "./types/ChainSpecifications";
+import {
+    DepositParams,
+    DepositETHParams,
+    BalanceOfParams,
+    ApproveDepositParams,
+    GetDepositAddressParams,
+} from "./parameters";
+import { parseUnits, formatUnits } from "viem";
+
+export class RenzoService {
+    @Tool({
+        name: "renzo_deposit_erc20",
+        description:
+            "Deposit ERC20 LST tokens into Renzo, approve the ERC20 contract to spend the tokens before calling this",
+    })
+    async depositERC20(
+        walletClient: EVMWalletClient,
+        parameters: DepositParams
+    ) {
+        try {
+            console.log(
```

Comment:
No `console.logs` in the plugins! Good for debugging but no need for that on the agent :)